pkg/generators/forwarding/fluentd/templates.go

```diff
@@ -684,11 +684,16 @@ const outputLabelConfJsonParseNoretryTemplate = `{{- define "outputLabelConfJson
 	  pod_info        ${if record['kubernetes_info'] != nil && record['kubernetes_info'] != {}; "pod_name=" + record['kubernetes_info']['pod_name']; else nil; end}
 	  container_info  ${if record['kubernetes_info'] != nil && record['kubernetes_info'] != {}; "container_name=" + record['kubernetes_info']['container_name']; else nil; end}
 	  msg_key         ${if record.has_key?('message') && record['message'] != nil; record['message']; else nil; end}
-      msg_info        ${if record['msg_key'] != nil && record['msg_key'].is_a?(Hash); require 'json'; "message="+record['message'].to_json; elsif record['msg_key'] != nil; "message="+record['message']; else nil; end}
-      message         ${if record['msg_key'] != nil && record['kubernetes_info'] != nil && record['kubernetes_info'] != {}; record['namespace_info'] + ", " + record['container_info'] + ", " + record['pod_info'] + ", " + record['msg_info']; else record['message']; end}
+	  msg_info        ${if record['msg_key'] != nil && record['msg_key'].is_a?(Hash); require 'json'; "message="+record['message'].to_json; elsif record['msg_key'] != nil; "message="+record['message']; else nil; end}
+	  message         ${if record['msg_key'] != nil && record['kubernetes_info'] != nil && record['kubernetes_info'] != {}; record['namespace_info'] + ", " + record['container_info'] + ", " + record['pod_info'] + ", " + record['msg_info']; else record['message']; end}
+	  systemd_info    ${if record.has_key?('systemd') && record['systemd']['t'].has_key?('PID'); record['systemd']['u']['SYSLOG_IDENTIFIER'] += "[" + record['systemd']['t']['PID'] + "]"; else {}; end}
 	</record>
-	remove_keys kubernetes_info, namespace_info, pod_info, container_info, msg_key, msg_info
+	remove_keys kubernetes_info, namespace_info, pod_info, container_info, msg_key, msg_info, systemd_info
   </filter>
+  <match journal.** system.var.log**>
+    @type copy
+{{include .StoreTemplate . "pick_daemon_name" | indent 4}}
+  </match>
```

Comment:
i dont catch using the syslog store template here.

Reply:
Infrastructure pod logs and journal logs are sent under "infrastructure".
But the infrastructure pod logs doesn't have $.systemd.u.SYSLOG_IDENTIFIER.
So I splitted "infrastructure" stream into the pod logs and the journal logs and then injected the daemon name into the journal logs only.
This syslog store template is for the journal logs. I will add some comments to explain this.